services/ui-src/src/components/reports/OverlayModalPage.tsx

```diff
@@ -23,24 +23,35 @@ import { EntityShape, OverlayModalPageShape } from "types";
 import { getFormattedEntityData, resetClearProp, useStore } from "utils";
 
 export const OverlayModalPage = ({
-  entity,
   closeEntityDetailsOverlay,
   route,
 }: Props) => {
-  const { entityType, verbiage, modalForm, stepType } = route;
-  const { selectedEntity: currentEntity } = useStore();
-  const [selectedEntity, setSelectedEntity] = useState<EntityShape | undefined>(
-    undefined
-  );
+  const { verbiage, modalForm, stepType } = route;
+  const { report, selectedEntity, setSelectedEntity } = useStore();
+  const [selectedStepEntity, setSelectedStepEntity] = useState<
+    EntityShape | undefined
+  >(undefined);
+
+  const entityType = selectedEntity!.type;
```

Comment:
This looks like it's doing more than it is. We just removed entityType, since it's now coming from selectedEntity, and cleaned up the naming scheme.